From 3819f5aef4438431bcf4f7655c6c49105c271aa7 Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Tue, 2 Aug 2022 18:21:01 -0700
Subject: [PATCH] iMX8M: add SPL-only build for DDR4

The flash.bin image therefore becomes the SPL image ready to be
signed.  This change is specific to DDR4 firmware.

Upstream-Status: Inappropriate [lmp specific]

Signed-off-by: Michael Scott <mike@foundries.io>
---
 iMX8M/soc.mak | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 4d48193..2c20376 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -112,7 +112,7 @@ ddr4_dmem_2d = ddr4_dmem_2d$(DDR_FW_VERSION).bin
 
 u-boot-spl-ddr4.bin: u-boot-spl.bin $(ddr4_imem_1d) $(ddr4_dmem_1d) $(ddr4_imem_2d) $(ddr4_dmem_2d)
 	@objcopy -I binary -O binary --pad-to 0x8000 --gap-fill=0x0 $(ddr4_imem_1d) ddr4_imem_1d_pad.bin
-	@objcopy -I binary -O binary --pad-to 0x4000 --gap-fill=0x0 $(ddr4_dmem_1d) ddr4_dmem_1d_pad.bin
+	@objcopy -I binary -O binary --pad-to 0x1000 --gap-fill=0x0 $(ddr4_dmem_1d) ddr4_dmem_1d_pad.bin
 	@objcopy -I binary -O binary --pad-to 0x8000 --gap-fill=0x0 $(ddr4_imem_2d) ddr4_imem_2d_pad.bin
 	@cat ddr4_imem_1d_pad.bin ddr4_dmem_1d_pad.bin > ddr4_1d_fw.bin
 	@cat ddr4_imem_2d_pad.bin $(ddr4_dmem_2d) > ddr4_2d_fw.bin
@@ -250,6 +250,9 @@ flash_evk_spl: $(MKIMG) u-boot-spl-combine u-boot-spl-ddr.bin
 	./mkimage_imx8 -version $(VERSION) -loader u-boot-spl-ddr.bin $(SPL_LOAD_ADDR) -out $(OUTIMG)
 endif
 
+flash_evk_spl_ddr4: $(MKIMG) u-boot-spl-combine u-boot-spl-ddr4.bin
+	./mkimage_imx8 -version $(VERSION) -loader u-boot-spl-ddr4.bin $(SPL_LOAD_ADDR) -out $(OUTIMG)
+
 flash_evk_no_hdmi: $(MKIMG) u-boot-spl-ddr.bin u-boot.itb
 	./mkimage_imx8 -version $(VERSION) -fit -loader u-boot-spl-ddr.bin $(SPL_LOAD_ADDR) -second_loader u-boot.itb 0x40200000 0x60000 -out $(OUTIMG)
 
-- 
2.37.1

